src/db.c

```diff
@@ -582,11 +586,15 @@ int dbGenericDelete(redisDb *db, robj *key, int async, int flags) {
             if (dictDelete(db->expires[slot],key->ptr) == DICT_OK) {
                 cumulativeKeyCountAdd(db, slot, -1, DB_EXPIRES);
                 db->sub_dict[DB_EXPIRES].key_count--;
+                if (dictSize(db->expires[slot]) == 0)
+                    db->sub_dict[DB_EXPIRES].non_empty_slots--;
```

Comment:
maybe it would be nicer if both `key_count` and `non_empty_slots` will be handled by `cumulativeKeyCountAdd`?
it means we have to modify the comment that says it only handles cluster slots related aspects, and it would mean we have to always call it **after** adding the key to the dict (otherwise the `dictSize` call will not be a valid check).
@guybe7 FYI, in the context of #12822

Reply:
I tried doing this in 8d3227f, we can compare which method is more suitable, whether inside or outside of `cumulativeKeyCountAdd`.

Reply:
thanks, that aligns better with https://github.com/redis/redis/pull/12822